src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -946,15 +946,10 @@ public Consumer createConsumerFromDTO(ConsumerDTO consumer, ConsumerType type, P
         // API:registerConsumer
         Set<String> keyStrings = splitKeys(activationKeys);
 
-        // Only let NoAuth principals through if there are activation keys to consider:
-        if ((principal instanceof NoAuthPrincipal) && keyStrings.isEmpty()) {
-            throw new ForbiddenException(i18n.tr("Insufficient permissions"));
-        }
-
         validateOnKeyStrings(keyStrings, owner.getKey(), userName);
```

Comment:
It might be needed. If the request got authenticated by any other method it would skip these validations. We can give activation key auth a higher priority, but I am not sure if that's something we want.